tstrait/simulate_effect_size.py

```diff
@@ -233,21 +227,44 @@ def sim_trait(ts, model, *, num_causal=None, alpha=None, random_seed=None):
     """
     ts = _check_instance(ts, "ts", tskit.TreeSequence)
     model = _check_instance(model, "model", tstrait.TraitModel)
-    num_causal = 1 if num_causal is None else num_causal
-    num_causal = _check_int(num_causal, "num_causal", minimum=1)
+    if num_causal is not None and causal_sites is not None:
+        raise ValueError("Cannot specify both num_causal and causal_sites")
+    if num_causal is None and causal_sites is None:
+        num_causal = 1
+
+    if isinstance(causal_sites, numbers.Real):
+        causal_sites = np.array([causal_sites])
+
+    if causal_sites is not None:
+        causal_sites = np.sort(causal_sites)
+        if len(causal_sites) > ts.num_sites:
```

Comment:
I don't think it's worth checking for this, you'll find bad sites later on anyway, right?

Reply:
Ah hmm, looks like you're not. You definitely have to check for out of bounds sites and repeated values. These need to be tested also.

the current check is much too weak and not very useful.

Reply:
We will get "IndexError: Index out of bounds" error in tskit, so I think we only need to check for repeated values.